### ISSUE TITLE
math.hypot overflow

### ISSUE THREAD

**Issue by MSeifert04** [numba/numba#1877]
Currently math.hypot inside `njit`ted functions overflows when using large inputs:

```
import math
@nb.njit
def hypot(a, b):
    return math.hypot(a, b)

hypot(1e303, 1e303)     # inf
math.hypot(1e303, 1e303) # 1.414213562373095e+303
```
I somehow remember reading that numba implements functions itself, so maybe it's linked against some naive implementation?

```
@nb.njit
def naive_hypot(a, b):
    return math.sqrt(a**2 + b**2)

naive_hypot(1e303, 1e303) # inf
```
instead of an overflow-aware form:

```
@nb.njit
def overflow_aware_hypot(a, b):
    if a < b:
        a, b = b, a
    # a will be the bigger one, take the absolute to avoid wrong sign
    if a < 0:
        a = -a
    return a * math.sqrt(1 + (b/a)**2)

overflow_aware_hypot(1e303, 1e303) # 1.4142135623730953e+303
```
at least something like this. (Notice that also the last digit doesn't match the true `math.hypot`-result.

**Comment by seibert** [comment]
This is related to open PR #1619.

**Comment by seibert** [comment]
I think the solution we want to move toward is using hypot from libm, which should implement the fully correct version.

**Comment by MSeifert04** [comment]
@seibert Sorry I thought I searched for hypot-related issues but I must have been in the wrong github project. Thank you for the answer.

**Comment by seibert** [comment]
Fixed by #1934.
